### ISSUE TITLE
TestTwoProcessesPosix.tearDown has a gratuitous print statement

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1921]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| spiv reported|
|-|-|
|Trac ID|trac#1921|
|Type|defect|
|Created|2006-07-13 18:10:48Z|

```python
    def tearDown(self):
        for i in (0,1):
            pp, process = self.pp[i], self.processes[i]
            if not pp.finished:
                try:
                    os.kill(process.pid, signal.SIGTERM)
                except OSError:
                    print "OSError"
        return self._onClose()
```

When the except clause is tripped, this test case prints a useless, uninformative message that messes up the normal terminal output from trial, and the message is hard to grep for.

The intent of this isn't clear.  If it's a real problem, it should fail the test.  If it's just debugging, then it should be logged in test.log rather than printed to the terminal (and probably logging the actual error message would be a good idea, too).

(While we're looking at this, it really ought to use `for pp, process in zip(self.pp, self.processes)`...)

Attachments:

* [test_process_1921.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/8dc/8dcb583b3ba5aa22850f93db0eec4c2a076d2cce/341d189836ee50fb91f50d113583d30d9cf3e37d.diff) (744 bytes) - added by therve on 2007-02-02 11:24:18Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__1921 1921
type__defect defect
reporter__spiv spiv
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1152814248000000 1152814248000000
changetime__1171033761000000 1171033761000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

To be explicit, I'm referring to code in twisted/test/test_process.py.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph set owner to spiv|
|-|-|

Trying to do some triage...

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve removed owner|
|-|-|

Less tickets make me happy.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set owner to @therve|
|-|-|

Maybe add comment explaining why we might get OSError: "if the test is correct the process should already be dead, so we'll always get OSError here for correct code." Assuming my interpretation of the code is correct...

Other than that looks good.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set status to `closed`|
|-|-|

(In [19592]) Author: therve
Reviewer: itamarst
Fixes [#1921](https://github.com/twisted/twisted/issues/1921)

Cleanup TestTwoProcessesPosix.tearDown: remove a print statement, use zip to
clarify code.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
